### PR TITLE
feat(btrfs): add btrfs support

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

File: control-plane/csi-driver/src/bin/node/filesystem_ops.rs [hunk 1]
```diff
@@ -290,6 +294,103 @@ impl FileSystemOps for XFs {
...
+    fn unmount_on_fs_id_diff(
+        &self,
+        device_path: &str,
+        fs_staging_path: &str,
+        volume_uuid: &Uuid,
+    ) -> Result<(), Error> {
+        mount::unmount_on_fs_id_diff(device_path, fs_staging_path, volume_uuid)
+    }
+
+    /// Btrfs filesystem needs an check.
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/node/filesystem_ops.rs]: What does needs a check mean?

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/csi-driver/src/bin/node/filesystem_ops.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/node/filesystem_ops.rs]: This check as per the docs is suggested before the metadata change, in read-only mode, this option exists to calm potential panic when users are going to run the checker.

Maybe the comment is not correct?

Comment (tiagolobocastro) [reply control-plane/csi-driver/src/bin/node/filesystem_ops.rs]: Yeah just looks a bit out of place to have a blanket statement like that on top of the function, I'd remove it or elaborate more on it in regards to repair.

Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/node/filesystem_ops.rs]: I don't think it does any repair(dangerous operation as termed by btrfs) it's just to allow us the uuid change. Should I remove the comment then?

Comment (tiagolobocastro) [reply control-plane/csi-driver/src/bin/node/filesystem_ops.rs]: Yeah or perhaps document that we don't do any repair then as it's dangerous and why.

Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/node/filesystem_ops.rs]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**